### ISSUE TITLE
What does it actually do?

### ISSUE THREAD

**Issue by f1am3d** [onnx/tensorflow-onnx#2232]
# Ask a Question

### Question
The description is unclear. 
1. What does this repository actually do? 
2. Can I use it to convert ckpt to onnx?
3. Can I use it to convert safetensors to onnx?
4. Can it be used for stable diffustion in generat?

I am asking because I used [this official AMD manual](https://community.amd.com/t5/gaming/how-to-running-optimized-automatic1111-stable-diffusion-webui-on/ba-p/625585) to convert SD to ONNX, but I need to convert custom model to ONNX and I am searching for a way how to do it.

**Comment by justinchuby** [comment]
For ckpt and safetensors, looks like you are looking for ways to convert wights (instead of model structures) to ONNX representation. You may consider https://github.com/justinchuby/onnx-safetensors for working with safetensors.

**Comment by Joly0** [comment]
@f1am3d Did you find a way to convert a safetensors file to onnx? I am trying to achieve the same as you, but even with the onnx-safetensors extension i am not able to convert a .safetensors file to onnx. Could you help me with that if you found a solution?

**Comment by justinchuby** [comment]
@Joly0 You can share the script you used here. If it is a bug in onnx-safetensors I am happy to fix that.

**Comment by fatcat-z** [comment]
> # What does it actually do?
> ### Question
> The description is unclear.
> 
> 1. What does this repository actually do?
> 2. Can I use it to convert ckpt to onnx?
> 3. Can I use it to convert safetensors to onnx?
> 4. Can it be used for stable diffustion in generat?
> 
> I am asking because I used [this official AMD manual](https://community.amd.com/t5/gaming/how-to-running-optimized-automatic1111-stable-diffusion-webui-on/ba-p/625585) to convert SD to ONNX, but I need to convert custom model to ONNX and I am searching for a way how to do it.

Yes, this tool aims to help TensorFlow users to convert their models to ONNX, no matter public models or custom models. 

Then answers to question 2) and 4) are Yes, as long as they are developed by TensorFlow.
